CMakeLists.txt

```diff
@@ -5,6 +5,10 @@ if(CMAKE_VERSION VERSION_EQUAL 3.0 OR CMAKE_VERSION VERSION_GREATER 3.0)
     cmake_policy(SET CMP0042 NEW)
 endif()
 
+if(POLICY CMP0058)
+  cmake_policy(SET CMP0058 NEW)
```

Comment:
With newer versions of cmake if the ninja option is set, A warning is generated if this policy if not set  one way or the other.

I was seeing this issue when I was trying debian sid unstable branch, so it may not be on any stable distros yet.

For stability reasons I went back to stock Ubuntu 16.04.  So I can no longer reproduce this issue as cmake ninja is not supported in stock Ubuntu 16.04